app/models/manageiq/providers/amazon/inventory/parser/cloud_manager.rb

```diff
@@ -4,10 +4,16 @@ class ManageIQ::Providers::Amazon::Inventory::Parser::CloudManager < ManageIQ::P
   def parse
     log_header = "MIQ(#{self.class.name}.#{__method__}) Collecting data for EMS name: [#{collector.manager.name}] id: [#{collector.manager.id}]"
     $aws_log.info("#{log_header}...")
+
     # The order of the below methods does matter, because they are searched using find instead of lazy_find
     flavors
 
+    # @mapper has to be initialized for parsing images & instances
+    # TODO: maybe do this in Persister?
+    @tag_mapper = ContainerLabelTagMapping.mapper
+
     # The order of the below methods doesn't matter since they refer to each other using only lazy links
+    persister.collections[:tags_to_resolve] = @tag_mapper.tags_to_resolve_collection
```

Comment:
cool, you already use custom IC here, I was about to ask about it :-)

Reply:
@cben could we move the IC definition into the defaults, and the init into the Persitors?

Reply:
I could.  Note that constructing the Mapper actually reads whole mappings table.
- Is it OK to do it each Persister initialization?
- Will it still be re-read each refresh? (eg. if it's only read once when refresh worker starts that would be bad, changing mapping would have no effect).  
- Conversely, can re-reading it each targetted refresh be a performance problem?
- Will Persister initialization remain a good place with collector/persister split?

Reply:
hum, ok, if it reads the whole mapping table, I will need to look on it more closely tomorrow. We will need to limit it only to Vms and Templates that are part of the targeted refresh

Reply:
- Is it OK to do it each Persister initialization?

Right now that would be the same as having it in parser, that is being called each refresh

- Will it still be re-read each refresh? (eg. if it's only read once when refresh worker starts that would be bad, changing mapping would have no effect).

Yes, right now, the persister defines ICs dynamically (that might change in the future, as some providers use static definition, but there can be always a dynamic part)

- Conversely, can re-reading it each targetted refresh be a performance problem?

Yes looking at this, it will definitely hit targeted refresh performance, with huge amount of tags present

- Will Persister initialization remain a good place with collector/persister split?

There will be some refactoring of the format, but the Persistor should remain mostly the same.